packages/react-client/src/ReactFlightClientContext.js

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+
+const {useContext, useMemo, createContext} = React;
+
+const ServerContextContext = createContext(null);
+
+type ServerContextNodeType = {
+  parent: null | ServerContextNodeType,
+  name: string,
+  value: any,
+};
+
+export function ServerContextWrapper({ServerContext, value, children}) {
```

Comment:
I'd put this in the FlightClient instead and just export the context there. That way all the complex logic is isolated there and you don't need to pull it in until a client actually loads.

Reply:
Similarly, if we did enable this on the Server too, then this implementation on the server might not be the same exactly.